kubernetes-client/src/com/goyeau/kubernetes/client/operation/Creatable.scala

```diff
@@ -50,8 +50,14 @@ private[client] trait Creatable[F[_], Resource <: { def metadata: Option[ObjectM
       .use(EnrichedStatus.apply[F])
       .flatMap {
         case status if status.isSuccess => update
-        case Status.NotFound            => create(resource).recoverWith { case Status.Conflict => update }
-        case status                     => F.pure(status)
+        case Status.NotFound =>
+          create(resource)
+            .flatMap {
+              case Status.Conflict => F.raiseError[Status](UnexpectedStatus(Status.Conflict, GET, fullResourceUri))
+              case status          => F.pure(status)
+            }
```

Comment:
Why did this change?

Reply:
The previous code was not correct and Scala 3 makes it not compile rightfully.

Reply:
Thanks for the explanation 👍 my question should have been "why did this change so extensively", the updated diff here looks more like what I would expect :)